ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -808,6 +815,12 @@ protected void fixMissingStoreKeys(ConnectedChannel connectedChannel, List<Remot
           getMessagesForMissingKeys(connectedChannel, exchangeMetadataResponseList, replicasToReplicatePerNode,
               remoteNode, remoteColoGetRequestForStandby);
       if (getResponse != null) {
+        if (getResponse.getError() != ServerErrorCode.No_Error) {
```

Comment:
nit: Similar as above. Looks like we are already checking for this error in `getMessagesForMissingKeys`.